nixos/modules/services/printing/lprint.nix

```diff
@@ -0,0 +1,506 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+let
+  inherit (lib) types;
+  magicUUID = "urn:uuid:4bd2d7e2-31d0-37d4-4399-08db6a587dd3"; # Arbitrary UUID, must remain constant
+  papplConfigOptions = {
+    dnssdName = lib.mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        The name of the service as it appears in DNS-SD.
+      '';
+    };
+    contact = lib.mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        The contact information for the service.
+      '';
+    };
+    maxImageSize = lib.mkOption {
+      type = types.submoduleWith {
+        modules = [
+          {
+            options = {
+              maxSize = lib.mkOption {
+                type = types.int;
+                default = 0;
+                description = ''
+                  The maximum image size in bytes.
+                '';
+              };
+              width = lib.mkOption {
+                type = types.int;
+                default = 0;
+                description = ''
+                  The maximum image width in pixels.
+                '';
+              };
+              height = lib.mkOption {
+                type = types.int;
+                default = 0;
+                description = ''
+                  The maximum image height in pixels.
+                '';
+              };
+            };
+          }
+        ];
+      };
+      description = ''
+        The maximum image size.
+      '';
+    };
+  };
+  globalPapplConfigType = types.submoduleWith {
+    modules = [
```

Comment:
why use `submoduleWith`
you're not adding specialArgs?

Reply:
Switched to submoduleWith when troubleshooting a hard-to-pin-down error. I'll switch back.